konduit-serving-core/src/main/java/ai/konduit/serving/pipeline/steps/BasePipelineStepRunner.java

```diff
@@ -22,70 +22,87 @@
 
 package ai.konduit.serving.pipeline.steps;
 
+import ai.konduit.serving.config.SchemaType;
 import ai.konduit.serving.pipeline.PipelineStep;
 import ai.konduit.serving.pipeline.PipelineStepRunner;
-import ai.konduit.serving.config.SchemaType;
 import org.datavec.api.records.Record;
-import org.datavec.api.writable.Writable;
+import org.datavec.api.writable.*;
+import org.nd4j.linalg.api.ndarray.INDArray;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
-public abstract  class BasePipelineStepRunner implements PipelineStepRunner {
+public abstract class BasePipelineStepRunner implements PipelineStepRunner {
```

Comment:
We might want to add some javadoc here at the top to make sure what base logic we have here.